homeassistant/components/sensor/fints.py

```diff
@@ -0,0 +1,288 @@
+"""
+Read the balance of your bank accounts via FinTS.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.fints/
+"""
+
+from collections import namedtuple
+from datetime import timedelta
+import logging
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_USERNAME, CONF_PIN, CONF_URL, CONF_NAME
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['fints==0.2.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(hours=4)
+
+ICON = 'mdi:currency-eur'
+
+BankCredentials = namedtuple('BankCredentials', 'blz login pin url')
+
+CONF_BIN = 'bank_identification_number'
+CONF_ACCOUNTS = 'accounts'
+CONF_HOLDINGS = 'holdings'
+CONF_ACCOUNT = 'account'
+
+ATTR_ACCOUNT = CONF_ACCOUNT
+ATTR_BANK = 'bank'
+ATTR_ACCOUNT_TYPE = 'account_type'
+
+SCHEMA_ACCOUNTS = vol.Schema({
+    vol.Required(CONF_ACCOUNT): cv.string,
+    vol.Optional(CONF_NAME, default=None): cv.string,
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_BIN): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PIN): cv.string,
+    vol.Required(CONF_URL): cv.string,
+    vol.Optional(CONF_NAME): cv.string,
+    vol.Optional(CONF_ACCOUNTS, default=[]): cv.ensure_list(SCHEMA_ACCOUNTS),
+    vol.Optional(CONF_HOLDINGS, default=[]): cv.ensure_list(SCHEMA_ACCOUNTS),
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the sensors.
+
+    Login to the bank and get a list of existing accounts. Create a
+    sensor for each account.
+    """
+    credentials = BankCredentials(config[CONF_BIN], config[CONF_USERNAME],
+                                  config[CONF_PIN], config[CONF_URL])
+    fints_name = config[CONF_NAME]
+
+    account_config = dict()
+    for acc in config[CONF_ACCOUNTS]:
+        account_config[acc[CONF_ACCOUNT]] = acc[CONF_NAME]
+
+    holdings_config = dict()
+    for acc in config[CONF_HOLDINGS]:
+        holdings_config[acc[CONF_ACCOUNT]] = acc[CONF_NAME]
+
+    client = FinTsClient(credentials, config[CONF_NAME])
+    balance_accounts, holdings_accounts = client.detect_accounts()
+    accounts = []
+
+    for account in balance_accounts:
+        if not config[CONF_ACCOUNTS] \
+                or account.iban in account_config:
+
+            account_name = account_config.get(account.iban)
+            if not account_name:
+                account_name = '{} - {}'.format(fints_name, account.iban)
+            accounts += [FinTsAccount(client, account, account_name)]
```

Comment:
`.append()`

Reply:
Your version creates a new list [] and the += likely .extends() the original list (extend loops over the new list and appends each item)

Dont have actual numbers but just append should be much more efficient